recipe-server/client/control/components/MultiPicker.js

```diff
@@ -51,88 +44,65 @@ export default class MultiPicker extends React.Component {
       value,
     } = this.props;
 
-    const selectedOptions = options.filter(option =>
-      value.indexOf(option.value) === -1);
-
-    let displayedOptions = [].concat(selectedOptions)
-      .map(option => option.value)
-      .map(val =>
-        (!val ? null : { ...this.convertValueToObj(val) }))
-      .filter(x => x);
-
     const {
       filterText,
     } = this.state;
 
-    if (filterText) {
-      displayedOptions = displayedOptions.filter(option =>
-        JSON.stringify(option).indexOf(filterText) > -1);
-    }
+    let selectedOptions = options.filter(option => value.indexOf(option.value) === -1);
 
-    return displayedOptions;
-  }
-
-  handleApplyOption(event) {
-    event.persist();
-
-    if (!this.availableRef) {
-      return;
-    }
-
-    const {
-      value,
-      onChange,
-    } = this.props;
-
-    let selectedFilters = MultiPicker.getActiveSelectOptions(this.availableRef);
+    if (filterText) {
+      const searchText = filterText.toLowerCase();
 
-    if (!selectedFilters || selectedFilters.length === 0) {
-      selectedFilters = this.getDisplayedOptions();
-      selectedFilters = selectedFilters.map(option => option.value);
+      selectedOptions = selectedOptions.filter(option => {
+        const searchValues = JSON.stringify(Object.values(option)).toLowerCase();
```

Comment:
I wanted to avoid hardcoding the values for it to search over. Does this suffice:

```
filteredOptions = filteredOptions.filter(({value, label}) =>{
  const searchValues = [value, label].join('   ').toLowerCase();
  return searchValues.indexOf(filterText) > -1;
});
```

or did you have other thoughts on how that should be searched over entirely?